src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java

```diff
@@ -384,6 +397,7 @@ public ResponseEntity<List<CatalogObjectMetadata>> list(
             @ApiParam(value = "Filter according to Content-Type.") @RequestParam(required = false) Optional<String> contentType,
             @ApiParam(value = "Filter according to Object Name.") @RequestParam(value = "objectName", required = false) Optional<String> objectNameFilter,
             @ApiParam(value = "Filter according to Object Tag.") @RequestParam(value = "objectTag", required = false) Optional<String> objectTagFilter,
+            @ApiParam(value = "Filter according to Job-Planner association status. If enabled, only objects for which a job-planner association exists with the provided status will be returned. Can be PLANNED, DEACTIVATED, FAILED or UNPLANNED") @RequestParam(value = "associationStatus", required = false) Optional<String> associationStatusFilter,
```

Comment:
@fviale According to Denis, we need a :"All" value for all 3 planning states. (Unplanned is for objects not being associated in JP at all.
Is this all state inferred from the absence of this query parameter ?

Reply:
@amouhoub the All value is exactly as you said, just don't add the filter.

Reply:
but yes, I did not implement a All that would only select objects with either of PLANNED, DEACTIVATED or FAILED. I did not know this was requested.